rust/kernel/types.rs

```diff
@@ -247,3 +247,52 @@ impl<T: FnOnce()> Drop for ScopeGuard<T> {
         }
     }
 }
+
+/// Trait for wrappers which are holding a direct pointer that they are getting from a C side function.
+pub trait SavedAsPointer {
```

Comment:
Is there a reason why you can't use `PointerWrapper`? We should unify all these pointer wrappers.

Reply:
My Understanding of PointerWrapper is that it is designed for rust types which can be given to a C interface. The trait I am implementing is designed to hold a pointer which got allocated via a C api.

Reply:
@wedsonaf is my understanding of `PointerWrapper` correct? if not, should I withdraw this pr?